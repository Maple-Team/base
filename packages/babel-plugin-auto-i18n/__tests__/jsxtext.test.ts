import { getTransformCode } from './test-help'

describe('arrow function declaration scenarios - JSXText cases', () => {
  describe('exported arrow function cases', () => {
    it('handle JSXText1', () => {
      const sourceCode = `
    export const Component = () => {
      return <div>你好</div>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText2', () => {
      const sourceCode = `
    export const Component = () => {
      return <div>你好<span>嵌套文本</span></div>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText3', () => {
      const sourceCode = `
    export const Component = () => {
      return <>
      <div>你好</div>
      <span>嵌套文本</span>
      </>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext3.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
  describe('exported default arrow function cases', () => {
    it('handle JSXText1', () => {
      const sourceCode = `
    export default () => {
      return <div>你好</div>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText2', () => {
      const sourceCode = `
      export default () => {
      return <div>你好<span>嵌套文本</span></div>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText3', () => {
      const sourceCode = `
      export default () => {
      return <>
      <div>你好</div>
      <span>嵌套文本</span>
      </>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext3.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
})

describe('function declaration scenarios - JSXText cases', () => {
  describe('exported function cases', () => {
    it('handle JSXText1', () => {
      const sourceCode = `
    export function Component(){
      return <div>你好</div>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext1.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText2', () => {
      const sourceCode = `
    export function Component(){
      return <div>你好<span>嵌套文本</span></div>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText3', () => {
      const sourceCode = `
    export function Component(){
      return <>
      <div>你好</div>
      <span>嵌套文本</span>
      </>
    }
    `
      const result = getTransformCode(sourceCode, 'jsxtext3.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })

  describe('exported default function cases', () => {
    it('handle JSXText1', () => {
      const sourceCode = `
      export default function Component(){
        return <div>你好</div>
      }
      `
      const result = getTransformCode(sourceCode, 'jsxtext4.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText2', () => {
      const sourceCode = `
      export default function Component(){
        return <div>你好<span>嵌套文本</span></div>
      }
      `
      const result = getTransformCode(sourceCode, 'jsxtext2.tsx')
      expect(result?.code).toMatchSnapshot()
    })
    it('handle JSXText3', () => {
      const sourceCode = `
      export default function Component(){
        return <>
        <div>你好</div>
        <span>嵌套文本</span>
        </>
      }
      `
      const result = getTransformCode(sourceCode, 'jsxtext3.tsx')
      expect(result?.code).toMatchSnapshot()
    })
  })
})
