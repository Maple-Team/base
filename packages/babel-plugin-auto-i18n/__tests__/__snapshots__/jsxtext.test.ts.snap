// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow function declaration scenarios - JSXText cases exported arrow function cases handle JSXText1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>;
};"
`;

exports[`arrow function declaration scenarios - JSXText cases exported arrow function cases handle JSXText2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}<span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span></div>;
};"
`;

exports[`arrow function declaration scenarios - JSXText cases exported arrow function cases handle JSXText3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var _reactI18next = require(\\"react-i18next\\");
var Component = exports.Component = function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <>
      <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>
      <span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span>
      </>;
};"
`;

exports[`arrow function declaration scenarios - JSXText cases exported default arrow function cases handle JSXText1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _default = exports[\\"default\\"] = function _default() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>;
};"
`;

exports[`arrow function declaration scenarios - JSXText cases exported default arrow function cases handle JSXText2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _default = exports[\\"default\\"] = function _default() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}<span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span></div>;
};"
`;

exports[`arrow function declaration scenarios - JSXText cases exported default arrow function cases handle JSXText3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;
var _reactI18next = require(\\"react-i18next\\");
var _default = exports[\\"default\\"] = function _default() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <>
      <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>
      <span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span>
      </>;
};"
`;

exports[`function declaration scenarios - JSXText cases exported default function cases handle JSXText1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = Component;
var _reactI18next = require(\\"react-i18next\\");
function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>;
}"
`;

exports[`function declaration scenarios - JSXText cases exported default function cases handle JSXText2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = Component;
var _reactI18next = require(\\"react-i18next\\");
function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}<span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span></div>;
}"
`;

exports[`function declaration scenarios - JSXText cases exported default function cases handle JSXText3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = Component;
var _reactI18next = require(\\"react-i18next\\");
function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <>
        <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>
        <span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span>
        </>;
}"
`;

exports[`function declaration scenarios - JSXText cases exported function cases handle JSXText1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = Component;
var _reactI18next = require(\\"react-i18next\\");
function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>;
}"
`;

exports[`function declaration scenarios - JSXText cases exported function cases handle JSXText2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = Component;
var _reactI18next = require(\\"react-i18next\\");
function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}<span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span></div>;
}"
`;

exports[`function declaration scenarios - JSXText cases exported function cases handle JSXText3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = Component;
var _reactI18next = require(\\"react-i18next\\");
function Component() {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
    t = _useTranslation.t;
  return <>
      <div data-i18n=\\"[\\\\\\"key_\\\\u4F60\\\\u597D\\\\\\"]\\">{t(\\"key_你好\\")}</div>
      <span data-i18n=\\"[\\\\\\"key_\\\\u5D4C\\\\u5957\\\\u6587\\\\u672C\\\\\\"]\\">{t(\\"key_嵌套文本\\")}</span>
      </>;
}"
`;
